### PR TITLE
BYO worker

### PR DESCRIPTION
**Pull request by tclose**
## Types of changes

- New feature (non-breaking change which adds functionality)

## Summary
Adds the ability to supply your own worker, i.e. one not already defined in the pydra core, without it needing to monkey patch the pydra.engine.workers.WORKERS dict

## Checklist

- [ x ] I have added tests to cover my changes (if necessary)

### CODE REVIEW

File: pydra/engine/submitter.py [hunk 1]
```diff
@@ -16,24 +17,31 @@
...
         """
         Initialize task submission.
 
         Parameters
         ----------
-        plugin : :obj:`str`
-            The identifier of the execution backend.
+        plugin : :obj:`str` or :obj:`pydra.engine.core.Worker`
+            Either the identifier of the execution backend or the backend itself.
             Default is ``cf`` (Concurrent Futures).
```
Comment (effigies) [inline pydra/engine/submitter.py:29]: Should document the kwargs.

```suggestion
        \*\*kwargs ...
```

File: pydra/engine/submitter.py [hunk 2]
```diff
@@ -16,24 +17,31 @@
...
 
         """
         self.loop = get_open_loop()
         self._own_loop = not self.loop.is_running()
-        self.plugin = plugin
-        try:
-            self.worker = WORKERS[self.plugin](**kwargs)
-        except KeyError:
-            raise NotImplementedError(f"No worker for {self.plugin}")
+        if isinstance(plugin, Worker):
```
Comment (effigies) [inline pydra/engine/submitter.py:33]: Do we want to allow people to pass uninitialized classes, as well as pre-initialized? Here's an approach that looks up a class for `str`, initializes a class for `type`:

```python
if isinstance(plugin, str):
    plugin = WORKERS.get(plugin)
    if plugin is None:
        raise NotImplementedError(f"No worker for {plugin}")
if isinstance(plugin, type):
    plugin = plugin(**kwargs)

self.worker = plugin
self.plugin = plugin.plugin_name
```

Given that we're allowing initialization options, it seems weird not to permit classes, to me. But I can also see a case for keeping the options fewer. Up to you.

Alternately, we could get rid of `kwargs`. People can use defaults or pass a fully initialized `Worker`.

**Review by effigies — APPROVED** [review]
I agree with this approach in general. Small comments for your consideration.

File: pydra/engine/submitter.py (hunk 1 above)
Comment (tclose) [reply pydra/engine/submitter.py:29]: done

File: pydra/engine/submitter.py (hunk 2 above)
Comment (tclose) [reply pydra/engine/submitter.py]: Have changed it to a class instead of an instance

**Pull request merged**